x-pack/plugins/security_solution/public/flyout/shared/components/flyout_history_row.tsx

```diff
@@ -0,0 +1,189 @@
+import { useEventDetails } from '../../document_details/shared/hooks/use_event_details';
+import { getField, getAlertTitle, getEventTitle } from '../../document_details/shared/utils';
+import { RulePanelKey } from '../../rule_details/right';
+import { UserPanelKey } from '../../entity_details/user_right';
+import { HostPanelKey } from '../../entity_details/host_right';
+import { NetworkPanelKey } from '../../network_details';
+import { useRuleDetails } from '../../rule_details/hooks/use_rule_details';
+import {
+  DOCUMENT_DETAILS_HISTORY_ROW_TEST_ID,
+  RULE_HISTORY_ROW_TEST_ID,
+  GENERIC_HISTORY_ROW_TEST_ID,
+} from './test_ids';
+
+export interface FlyoutHistoryRowProps {
+  /**
+   * Flyout item to display
+   */
+  item: FlyoutPanelProps;
+  /**
+   * Index of the flyout in the list
+   */
+  index: number;
+}
+
+/**
+ * Row item for a flyout history row
+ */
+export const FlyoutHistoryRow: FC<FlyoutHistoryRowProps> = memo(({ item, index }) => {
+  switch (item.id) {
+    case DocumentDetailsRightPanelKey:
+      return <DocumentDetailsHistoryRow item={item} index={index} />;
+    case RulePanelKey:
+      return <RuleHistoryRow item={item} index={index} />;
+    case HostPanelKey:
+      return (
+        <GenericHistoryRow
+          item={item}
+          index={index}
+          title={String(item?.params?.hostName)}
+          icon={'storage'}
+          name={'Host'}
+        />
+      );
+    case UserPanelKey:
+      return (
+        <GenericHistoryRow
+          item={item}
+          index={index}
+          title={String(item?.params?.userName)}
+          icon={'user'}
+          name={'User'}
+        />
+      );
+    case NetworkPanelKey:
+      return (
+        <GenericHistoryRow
+          item={item}
+          index={index}
+          title={String(item?.params?.ip)}
+          icon={'globe'}
+          name={'Network'}
+        />
+      );
+  }
+  return null;
+});
+
+/**
+ * Row item for a document details
+ */
+export const DocumentDetailsHistoryRow: FC<FlyoutHistoryRowProps> = memo(({ item, index }) => {
+  const { openFlyout } = useExpandableFlyoutApi();
+  const { dataFormattedForFieldBrowser, getFieldsData } = useEventDetails({
+    eventId: String(item?.params?.id),
+    indexName: String(item?.params?.indexName),
+  });
+  const { ruleId, isAlert } = useBasicDataFromDetailsData(dataFormattedForFieldBrowser);
+  const { rule: maybeRule } = useRuleWithFallback(ruleId);
+  const eventKind = getField(getFieldsData('event.kind'));
+  const eventCategory = getField(getFieldsData('event.category'));
+
+  const title = useMemo(
+    () =>
+      isAlert
+        ? getAlertTitle({ ruleName: maybeRule?.name })
+        : getEventTitle({ eventKind, eventCategory, getFieldsData }),
+    [isAlert, maybeRule, eventKind, eventCategory, getFieldsData]
+  );
+
+  const onClick = useCallback(() => {
+    openFlyout({ right: item });
+  }, [openFlyout, item]);
+
+  return (
```

Comment:
any reason this and the rule below can't use the `GenericHistoryRow` component?

Reply:
do you mean using the `GenericHistoryRow` instead of `EuiContextMenuItem`?

Reply:
Yeah that's what I meant. This is really not a bit deal at all, I was just thinking if we could simplify things even more, but I'm not wondering if my suggestion could actually make the code more complex 😆

Reply:
no i think this is a good idea, actually make it more clean. i made it work by adding a `data-test-subj` prop, let me know what you think!

Reply:
I like it, this is what I had in mind, thanks!